client/app/pages/alert/Alert.jsx

```diff
@@ -0,0 +1,392 @@
+    const { alert } = this.state;
+    this.setState({ canceling: true });
+    navigateTo(`/alerts/${alert.id}`, true);
+  };
+
+  delete = () => {
+    const { alert } = this.state;
+
+    const doDelete = () => {
+      alert.$delete(() => {
+        notification.success('Alert deleted successfully.');
+        navigateTo('/alerts', true);
+      }, () => {
+        notification.error('Failed deleting alert.');
+      });
+    };
+
+    Modal.confirm({
+      title: 'Delete Alert',
+      content: 'Are you sure you want to delete this alert?',
+      okText: 'Delete',
+      okType: 'danger',
+      onOk: doDelete,
+      maskClosable: true,
+      autoFocusButton: null,
+    });
+  }
+
+  render() {
+    const { alert } = this.state;
+    if (!alert) {
+      return <LoadingState className="m-t-30" />;
+    }
+
+    const isNew = isNewAlert();
+    const { query, name, options } = alert;
+    const { queryResult, editMode, pendingRearm, canEdit, saving, canceling } = this.state;
+
+    return (
+      <div className="container alert-page">
+        <div className="p-b-10 m-l-0 m-r-0 page-header--new">
+          <div className="d-flex">
+            <h3>
+              {editMode && query ? (
+                <Input className="f-inherit" placeholder={this.getDefaultName()} value={name} onChange={e => this.setName(e.target.value)} />
+              ) : name || this.getDefaultName() }
+            </h3>
+            <span className="alert-actions">
+              {editMode && (
+                <>
+                  {!isNew && (
+                    <>
+                      <Button className="m-r-5" onClick={() => this.cancel()}>
+                        {canceling ? spinnerIcon : <i className="fa fa-times m-r-5" />}
+                        Cancel
+                      </Button>
+                      <Button type="primary" onClick={() => this.save()}>
+                        {saving ? spinnerIcon : <i className="fa fa-check m-r-5" />}
+                        Save Changes
+                      </Button>
+                    </>
+                  )}
+                </>
+              )}
+              {!editMode && canEdit && (
+                <Button type="default" onClick={() => this.edit()}><i className="fa fa-edit m-r-5" />Edit</Button>
+              )}
+              {canEdit && !isNew && (
+                <Dropdown
+                  className="m-l-5"
+                  trigger={['click']}
+                  placement="bottomRight"
+                  overlay={(
+                    <Menu>
+                      <Menu.Item>
+                        <a onClick={() => this.delete()}>Delete Alert</a>
+                      </Menu.Item>
+                    </Menu>
+                  )}
+                >
+                  <Button><Icon type="ellipsis" rotate={90} /></Button>
+                </Dropdown>
+              )}
+            </span>
+          </div>
+        </div>
+        <div className="row bg-white tiled p-20">
```

Comment:
GitHub is broken :grin:

Reply:
Heh heh... I think it's unfamiliar with fragment short syntax
```jsx
<>
  What's this?
</>
```

Reply:
Yeah :grin: